### PR TITLE
Feature/better archiving

### PR DESCRIPTION
**Pull request by southeo**
Archive operation now brought in line with FDO specifications. An archived PID has attributes 0-39, 100-101, as outlined in the RFC document. 

The archive operation is conducted as follows:
1. Generate tombstone attributes to be posted (e.g. tombstoneText, pidStatus = "ARCHIVED", etc.)
2. post/update existing handle record with generated attributes. Increment version. 
3. Delete any rows outside tombstone range [0-39, 100-101]

Note that the 10320/loc is meant to point to a landing page. There is currently a placeholder value there.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -70,10 +70,6 @@
       <groupId>org.springframework</groupId>
       <artifactId>spring-context</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
```
Comment (samleeflang) [inline pom.xml:75]: Is this correct? Can this be removed

**Review by samleeflang — COMMENTED** [review]
Nice changes, couple of questions

File: src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java [hunk 2]
```diff
@@ -147,6 +153,21 @@ public ResponseEntity<JsonApiWrapperWrite> updateRecords(@RequestBody List<JsonN
...
   }
 
+  // Upsert
+  @Operation(summary="Create a PID Record; if it already exists, update contents. DigitalSpecimens only.")
+  @PatchMapping(value="/upsert")
+  public ResponseEntity<JsonApiWrapperWrite> upsertRecord(List<JsonNode> requests)
+      throws InvalidRequestException, UnprocessableEntityException, PidResolutionException, PidServiceInternalError, JsonProcessingException {
+    for (var request: requests){
+      schemaValidator.validatePostRequest(request);
+      if (!request.get(NODE_DATA).get(NODE_TYPE).asText().equals(ObjectType.DIGITAL_SPECIMEN.toString())){
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/controller/HandleController.java:163]: The schemavalidator.validatePostRequest() function takes care of that

File: pom.xml (hunk 1 above)
Comment (southeo) [reply pom.xml:75]: It was a duplicate

File: src/main/java/eu/dissco/core/handlemanager/repository/HandleRepository.java [hunk 3]
```diff
@@ -215,23 +181,43 @@ private List<Query> prepareBatchUpdateQuery(long recordTimestamp, List<List<Hand
...
+          .set(HANDLES.TYPE, handleAttribute.type().getBytes(StandardCharsets.UTF_8))
+          .set(HANDLES.DATA, handleAttribute.data())
+          .set(HANDLES.TTL, TTL)
+          .set(HANDLES.TIMESTAMP, recordTimestamp)
+          .set(HANDLES.ADMIN_READ, true)
+          .set(HANDLES.ADMIN_WRITE, true)
+          .set(HANDLES.PUB_READ, true)
+          .set(HANDLES.PUB_WRITE, false);
       queryList.add(query);
+      if (updatedHandles.add(handleAttribute.handle())) {
```
[the remark this answers is not in the document]
Comment (southeo) [reply src/main/java/eu/dissco/core/handlemanager/repository/HandleRepository.java:217]: updatedHandles is a set i.e. unique elements only. `.add()` returns true if this set did not already contain the specified element.

[the remark this answers is not in the document]
Comment (samleeflang) [reply src/main/java/eu/dissco/core/handlemanager/repository/HandleRepository.java:217]: ok thanks

**Review by samleeflang — APPROVED** [review]

**Pull request merged**